tekton/integration_pipeline.go

```diff
@@ -125,15 +152,97 @@ func NewIntegrationPipelineRun(prefix, namespace string, integrationTestScenario
 		Spec: tektonv1.PipelineRunSpec{
 			PipelineRef: &tektonv1.PipelineRef{
 				ResolverRef: tektonv1.ResolverRef{
-					Resolver: tektonv1.ResolverName(integrationTestScenario.Spec.ResolverRef.Resolver),
-					Params:   resolverParams,
+					Resolver: tektonv1.ResolverName(resolver),
+					Params:   tektonResolverParams,
 				},
 			},
 		},
 	}
+
 	return &IntegrationPipelineRun{pipelineRun}
 }
 
+func getPipelineRunYamlFromPipelineRunResolver(client client.Client, ctx context.Context, prefix, namespace, resolver string, resolverParams []v1beta2.ResolverParameter) (string, error) {
+	stringResolverParams := map[string]string{}
+
+	for _, scenarioParam := range resolverParams {
+		stringResolverParams[scenarioParam.Name] = scenarioParam.Value
+	}
+
+	request := resolutionv1alpha1.ResolutionRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: prefix + "-",
+			Namespace:    namespace,
+			Labels: map[string]string{
+				"resolution.tekton.dev/type": resolver,
+			},
+			Annotations: map[string]string{
+				"konflux-ci.dev/created-by": "integration-service", // for backup garbage collection
```

Comment:
I like this idea, but it might work better as a label since labels are indexed and queryable, so this would make the job of any garbage collectors much easier.

Reply:
I used an annotation because they're lighter weight and because KubeArchive works with either. I can switch over to a label though in case we use some other mechanism for GC